inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/workflow/group/listener/UpdateGroupCompleteListener.java

```diff
@@ -49,35 +51,55 @@ public ProcessEvent event() {
     }
 
     @Override
-    public ListenerResult listen(WorkflowContext context) throws Exception {
+    public ListenerResult listen(WorkflowContext context) {
         GroupResourceProcessForm form = (GroupResourceProcessForm) context.getProcessForm();
         String groupId = form.getInlongGroupId();
         GroupOperateType operateType = form.getGroupOperateType();
         log.info("begin to execute UpdateGroupCompleteListener for groupId={}, operateType={}", groupId, operateType);
 
+        // update inlong group status and other configs
         String operator = context.getOperator();
         switch (operateType) {
             case SUSPEND:
                 groupService.updateStatus(groupId, GroupStatus.SUSPENDED.getCode(), operator);
-                sourceService.updateStatus(groupId, null, SourceStatus.SOURCE_FROZEN.getCode(), operator);
                 break;
             case RESTART:
                 groupService.updateStatus(groupId, GroupStatus.RESTARTED.getCode(), operator);
-                sourceService.updateStatus(groupId, null, SourceStatus.SOURCE_NORMAL.getCode(), operator);
                 break;
             case DELETE:
                 groupService.updateStatus(groupId, GroupStatus.DELETED.getCode(), operator);
-                sourceService.logicDeleteAll(groupId, null, operator);
                 break;
             default:
+                log.warn("Unsupported operate={} for inlong group", operateType);
                 break;
         }
+        InlongGroupInfo groupInfo = form.getGroupInfo();
+        groupService.update(groupInfo.genRequest(), operator);
 
-        // update inlong group status and other configs
-        groupService.update(form.getGroupInfo().genRequest(), operator);
+        // if the inlong group is lightweight mode, the stream source needs to be processed.
+        if (InlongConstants.LIGHTWEIGHT_MODE.equals(groupInfo.getLightweight())) {
+            changeSource4Lightweight(groupId, operateType, operator);
+        }
```

Comment:
Maybe LIGHTWEIGHT_MODE  need not use stream source

Reply:
The extract nodes for Sort were saved as stream sources. So, needs to operate especially in lightweight mode.